src/ut/sproutconnection_test.cpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * @file sproutconnection_test.cpp UT for SproutConnection
+ *
+ * Copyright (C) Metaswitch Networks 2018
+ * If license terms are provided to you in a COPYING file in the root directory
+ * of the source code repository by which you are accessing this code, then
+ * the license outlined in that COPYING file applies to your use.
+ * Otherwise no rights are granted except for those provided to you by
+ * Metaswitch Networks in a separate written agreement.
+ */
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "sproutconnection.h"
+#include "mock_httpclient.h"
+
+using ::testing::Mock;
+using ::testing::Return;
+using ::testing::_;
+using ::testing::AllOf;
+
+const SAS::TrailId FAKE_TRAIL_ID = 0x12345678;
+
+class SproutConnectionTest : public testing::Test
+{
+  static const std::string IMPU;
+  static const std::string IMPI;
+  static const std::vector<std::string> IMPIS;
+  static const std::vector<std::string> IMPUS;
+  static const std::string IMS_SUBSCRIPTION;
+  static const std::string dereg_body;
+  static const std::string dereg_body_no_impis;
+  static const std::string change_ids_body;
+
+  SproutConnectionTest()
+  {
+    _mock_http_client = new MockHttpClient();
+    HttpConnection* conn = new HttpConnection("server", _mock_http_client);
+    _sprout_conn = new SproutConnection(conn);
+
+    // If we don't override the default behaviour, return a nonsensical HTTP Code
+    ON_CALL(*_mock_http_client, send_request(_)).WillByDefault(Return(HttpResponse(-1, "", {})));
```

Comment:
Not sure i've put this in everywhere in my tests, which i should check, but should we actually be setting this up in the mock class somehow, rather than expecting users to just know to do this in any new future tests we write?

Reply:
I actually quite like that the author of the UTs has to make a conscious decision to add a catch-everything `ON_CALL` method if they want it. Hopefully, it makes people think about whether they really want it, or whether they actually care about the result in every individual UT